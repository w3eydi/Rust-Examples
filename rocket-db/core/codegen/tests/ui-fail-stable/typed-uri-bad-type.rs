../ui-fail/typed-uri-bad-type.rs