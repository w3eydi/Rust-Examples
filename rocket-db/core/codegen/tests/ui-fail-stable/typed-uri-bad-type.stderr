error[E0271]: type mismatch resolving `<std::string::String as FromParam<'_>>::Error == &str`
  --> $DIR/typed-uri-bad-type.rs:22:37
   |
22 | fn optionals(id: Option<i32>, name: Result<String, &str>) {  }
   |                                     ^^^^^^ expected enum `Infallible`, found `&str`

error[E0277]: the trait bound `usize: FromUriParam<rocket::http::uri::fmt::Path, &str>` is not satisfied
  --> $DIR/typed-uri-bad-type.rs:45:22
   |
45 |     uri!(simple(id = "hi"));
   |                      ^^^^ the trait `FromUriParam<rocket::http::uri::fmt::Path, &str>` is not implemented for `usize`
   |
   = help: the following implementations were found:
             <usize as FromUriParam<P, &'x mut usize>>
             <usize as FromUriParam<P, &'x usize>>
             <usize as FromUriParam<P, usize>>
   = note: required by `from_uri_param`

error[E0277]: the trait bound `usize: FromUriParam<rocket::http::uri::fmt::Path, &str>` is not satisfied
  --> $DIR/typed-uri-bad-type.rs:47:17
   |
47 |     uri!(simple("hello"));
   |                 ^^^^^^^ the trait `FromUriParam<rocket::http::uri::fmt::Path, &str>` is not implemented for `usize`
   |
   = help: the following implementations were found:
             <usize as FromUriParam<P, &'x mut usize>>
             <usize as FromUriParam<P, &'x usize>>
             <usize as FromUriParam<P, usize>>
   = note: required by `from_uri_param`

error[E0277]: the trait bound `usize: FromUriParam<rocket::http::uri::fmt::Path, i64>` is not satisfied
  --> $DIR/typed-uri-bad-type.rs:49:22
   |
49 |     uri!(simple(id = 239239i64));
   |                      ^^^^^^^^^ the trait `FromUriParam<rocket::http::uri::fmt::Path, i64>` is not implemented for `usize`
   |
   = help: the following implementations were found:
             <usize as FromUriParam<P, &'x mut usize>>
             <usize as FromUriParam<P, &'x usize>>
             <usize as FromUriParam<P, usize>>
   = note: required by `from_uri_param`

error[E0277]: the trait bound `S: FromUriParam<rocket::http::uri::fmt::Path, _>` is not satisfied
  --> $DIR/typed-uri-bad-type.rs:51:30
   |
51 |     uri!(not_uri_display(10, S));
   |                              ^ the trait `FromUriParam<rocket::http::uri::fmt::Path, _>` is not implemented for `S`
   |
   = note: required by `from_uri_param`

error[E0277]: the trait bound `i32: FromUriParam<rocket::http::uri::fmt::Path, std::option::Option<{integer}>>` is not satisfied
  --> $DIR/typed-uri-bad-type.rs:56:25
   |
56 |     uri!(optionals(id = Some(10), name = Ok("bob".into())));
   |                         ^^^^ the trait `FromUriParam<rocket::http::uri::fmt::Path, std::option::Option<{integer}>>` is not implemented for `i32`
   |
   = help: the following implementations were found:
             <i32 as FromUriParam<P, &'x i32>>
             <i32 as FromUriParam<P, &'x mut i32>>
             <i32 as FromUriParam<P, i32>>
   = note: required because of the requirements on the impl of `FromUriParam<rocket::http::uri::fmt::Path, std::option::Option<{integer}>>` for `std::option::Option<i32>`
   = note: required by `from_uri_param`

error[E0277]: the trait bound `std::string::String: FromUriParam<rocket::http::uri::fmt::Path, Result<_, _>>` is not satisfied
  --> $DIR/typed-uri-bad-type.rs:56:42
   |
56 |     uri!(optionals(id = Some(10), name = Ok("bob".into())));
   |                                          ^^ the trait `FromUriParam<rocket::http::uri::fmt::Path, Result<_, _>>` is not implemented for `std::string::String`
   |
   = help: the following implementations were found:
             <std::string::String as FromUriParam<P, &'a str>>
             <std::string::String as FromUriParam<P, &'x &'a str>>
             <std::string::String as FromUriParam<P, &'x mut &'a str>>
             <std::string::String as FromUriParam<P, &'x mut std::string::String>>
           and 2 others
   = note: required because of the requirements on the impl of `FromUriParam<rocket::http::uri::fmt::Path, Result<_, _>>` for `Result<std::string::String, &str>`
   = note: required by `from_uri_param`

error[E0277]: the trait bound `isize: FromUriParam<rocket::http::uri::fmt::Query, &str>` is not satisfied
  --> $DIR/typed-uri-bad-type.rs:58:19
   |
58 |     uri!(simple_q("hi"));
   |                   ^^^^ the trait `FromUriParam<rocket::http::uri::fmt::Query, &str>` is not implemented for `isize`
   |
   = help: the following implementations were found:
             <isize as FromUriParam<P, &'x isize>>
             <isize as FromUriParam<P, &'x mut isize>>
             <isize as FromUriParam<P, isize>>
   = note: required by `from_uri_param`

error[E0277]: the trait bound `isize: FromUriParam<rocket::http::uri::fmt::Query, &str>` is not satisfied
  --> $DIR/typed-uri-bad-type.rs:60:24
   |
60 |     uri!(simple_q(id = "hi"));
   |                        ^^^^ the trait `FromUriParam<rocket::http::uri::fmt::Query, &str>` is not implemented for `isize`
   |
   = help: the following implementations were found:
             <isize as FromUriParam<P, &'x isize>>
             <isize as FromUriParam<P, &'x mut isize>>
             <isize as FromUriParam<P, isize>>
   = note: required by `from_uri_param`

error[E0277]: the trait bound `S: FromUriParam<rocket::http::uri::fmt::Query, _>` is not satisfied
  --> $DIR/typed-uri-bad-type.rs:62:23
   |
62 |     uri!(other_q(100, S));
   |                       ^ the trait `FromUriParam<rocket::http::uri::fmt::Query, _>` is not implemented for `S`
   |
   = note: required by `from_uri_param`

error[E0277]: the trait bound `S: FromUriParam<rocket::http::uri::fmt::Query, _>` is not satisfied
  --> $DIR/typed-uri-bad-type.rs:64:25
   |
64 |     uri!(other_q(rest = S, id = 100));
   |                         ^ the trait `FromUriParam<rocket::http::uri::fmt::Query, _>` is not implemented for `S`
   |
   = note: required by `from_uri_param`

error[E0277]: the trait bound `S: Ignorable<rocket::http::uri::fmt::Query>` is not satisfied
   --> $DIR/typed-uri-bad-type.rs:66:25
    |
66  |     uri!(other_q(rest = _, id = 100));
    |                         ^ the trait `Ignorable<rocket::http::uri::fmt::Query>` is not implemented for `S`
    |
   ::: $WORKSPACE/core/http/src/uri/fmt/uri_display.rs
    |
    | pub fn assert_ignorable<P: Part, T: Ignorable<P>>() {  }
    |                                     ------------ required by this bound in `assert_ignorable`

error[E0277]: the trait bound `usize: Ignorable<rocket::http::uri::fmt::Query>` is not satisfied
   --> $DIR/typed-uri-bad-type.rs:68:33
    |
68  |     uri!(other_q(rest = S, id = _));
    |                                 ^ the trait `Ignorable<rocket::http::uri::fmt::Query>` is not implemented for `usize`
    |
   ::: $WORKSPACE/core/http/src/uri/fmt/uri_display.rs
    |
    | pub fn assert_ignorable<P: Part, T: Ignorable<P>>() {  }
    |                                     ------------ required by this bound in `assert_ignorable`

error[E0277]: the trait bound `S: FromUriParam<rocket::http::uri::fmt::Query, _>` is not satisfied
  --> $DIR/typed-uri-bad-type.rs:68:25
   |
68 |     uri!(other_q(rest = S, id = _));
   |                         ^ the trait `FromUriParam<rocket::http::uri::fmt::Query, _>` is not implemented for `S`
   |
   = note: required by `from_uri_param`

error[E0277]: the trait bound `usize: FromUriParam<rocket::http::uri::fmt::Path, &str>` is not satisfied
  --> $DIR/typed-uri-bad-type.rs:77:40
   |
77 |     uri!(uri!("?foo#bar"), simple(id = "hi"));
   |                                        ^^^^ the trait `FromUriParam<rocket::http::uri::fmt::Path, &str>` is not implemented for `usize`
   |
   = help: the following implementations were found:
             <usize as FromUriParam<P, &'x mut usize>>
             <usize as FromUriParam<P, &'x usize>>
             <usize as FromUriParam<P, usize>>
   = note: required by `from_uri_param`

error[E0277]: the trait bound `rocket::http::uri::Reference<'_>: ValidRoutePrefix` is not satisfied
  --> $DIR/typed-uri-bad-type.rs:77:15
   |
77 |     uri!(uri!("?foo#bar"), simple(id = "hi"));
   |               ^^^^^^^^^^ the trait `ValidRoutePrefix` is not implemented for `rocket::http::uri::Reference<'_>`

error[E0277]: the trait bound `usize: FromUriParam<rocket::http::uri::fmt::Path, &str>` is not satisfied
  --> $DIR/typed-uri-bad-type.rs:78:33
   |
78 |     uri!(uri!("*"), simple(id = "hi"));
   |                                 ^^^^ the trait `FromUriParam<rocket::http::uri::fmt::Path, &str>` is not implemented for `usize`
   |
   = help: the following implementations were found:
             <usize as FromUriParam<P, &'x mut usize>>
             <usize as FromUriParam<P, &'x usize>>
             <usize as FromUriParam<P, usize>>
   = note: required by `from_uri_param`

error[E0277]: the trait bound `rocket::http::uri::Asterisk: ValidRoutePrefix` is not satisfied
  --> $DIR/typed-uri-bad-type.rs:78:15
   |
78 |     uri!(uri!("*"), simple(id = "hi"));
   |               ^^^ the trait `ValidRoutePrefix` is not implemented for `rocket::http::uri::Asterisk`

error[E0277]: the trait bound `usize: FromUriParam<rocket::http::uri::fmt::Path, &str>` is not satisfied
  --> $DIR/typed-uri-bad-type.rs:81:25
   |
81 |     uri!(_, simple(id = "hi"), uri!("*"));
   |                         ^^^^ the trait `FromUriParam<rocket::http::uri::fmt::Path, &str>` is not implemented for `usize`
   |
   = help: the following implementations were found:
             <usize as FromUriParam<P, &'x mut usize>>
             <usize as FromUriParam<P, &'x usize>>
             <usize as FromUriParam<P, usize>>
   = note: required by `from_uri_param`

error[E0277]: the trait bound `rocket::http::uri::Asterisk: ValidRouteSuffix<rocket::http::uri::Origin<'static>>` is not satisfied
  --> $DIR/typed-uri-bad-type.rs:81:37
   |
81 |     uri!(_, simple(id = "hi"), uri!("*"));
   |                                     ^^^ the trait `ValidRouteSuffix<rocket::http::uri::Origin<'static>>` is not implemented for `rocket::http::uri::Asterisk`

error[E0277]: the trait bound `usize: FromUriParam<rocket::http::uri::fmt::Path, &str>` is not satisfied
  --> $DIR/typed-uri-bad-type.rs:82:25
   |
82 |     uri!(_, simple(id = "hi"), uri!("/foo/bar"));
   |                         ^^^^ the trait `FromUriParam<rocket::http::uri::fmt::Path, &str>` is not implemented for `usize`
   |
   = help: the following implementations were found:
             <usize as FromUriParam<P, &'x mut usize>>
             <usize as FromUriParam<P, &'x usize>>
             <usize as FromUriParam<P, usize>>
   = note: required by `from_uri_param`

error[E0277]: the trait bound `rocket::http::uri::Origin<'_>: ValidRouteSuffix<rocket::http::uri::Origin<'static>>` is not satisfied
  --> $DIR/typed-uri-bad-type.rs:82:37
   |
82 |     uri!(_, simple(id = "hi"), uri!("/foo/bar"));
   |                                     ^^^^^^^^^^ the trait `ValidRouteSuffix<rocket::http::uri::Origin<'static>>` is not implemented for `rocket::http::uri::Origin<'_>`
